src/main/start-main-application/start-main-application.injectable.ts

```diff
@@ -0,0 +1,76 @@
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+
+import { runManyFor } from "../../common/runnable/run-many-for";
+import { runManySyncFor } from "../../common/runnable/run-many-sync-for";
+import { beforeElectronIsReadyInjectionToken } from "./runnable-tokens/before-electron-is-ready-injection-token";
+import { beforeApplicationIsLoadingInjectionToken } from "./runnable-tokens/before-application-is-loading-injection-token";
+import { whenApplicationIsLoadingInjectionToken } from "./runnable-tokens/when-application-is-loading-injection-token";
+import { afterApplicationIsLoadedInjectionToken } from "./runnable-tokens/after-application-is-loaded-injection-token";
+import applicationIsLoadingWindowInjectable from "./lens-window/application-is-loading-window/application-is-loading-window.injectable";
+
+import applicationWindowInjectable from "./lens-window/application-window/application-window.injectable";
+import shouldStartHiddenInjectable from "../electron-app/features/should-start-hidden.injectable";
+import openDeepLinkInjectable from "../protocol-handler/lens-protocol-router-main/open-deep-link-for-url/open-deep-link.injectable";
+import { pipeline } from "@ogre-tools/fp";
+import { find, map, startsWith, toLower } from "lodash/fp";
+import commandLineArgumentsInjectable from "../utils/command-line-arguments.injectable";
+import waitForElectronToBeReadyInjectable from "../electron-app/features/wait-for-electron-to-be-ready.injectable";
+
+const startMainApplicationInjectable = getInjectable({
+  id: "start-main-application",
+
+  instantiate: (di) => {
+    const runMany = runManyFor(di);
+    const runManySync = runManySyncFor(di);
+    const waitForElectronToBeReady = di.inject(waitForElectronToBeReadyInjectable);
+    const applicationWindow = di.inject(applicationWindowInjectable);
+    const applicationIsLoadingWindow = di.inject(applicationIsLoadingWindowInjectable);
+    const shouldStartHidden = di.inject(shouldStartHiddenInjectable);
+    const openDeepLink = di.inject(openDeepLinkInjectable);
+    const commandLineArguments = di.inject(commandLineArgumentsInjectable);
+
+    const beforeElectronIsReady = runManySync(beforeElectronIsReadyInjectionToken);
+    const beforeApplicationIsLoading = runMany(beforeApplicationIsLoadingInjectionToken);
+    const whenApplicationIsLoading = runMany(whenApplicationIsLoadingInjectionToken);
+    const afterApplicationIsLoaded = runMany(afterApplicationIsLoadedInjectionToken);
+
+    return async () => {
+      // Stuff happening before application is ready needs to be synchronous because of
+      // https://github.com/electron/electron/issues/21370
+      beforeElectronIsReady();
+
+      await waitForElectronToBeReady();
+
+      await beforeApplicationIsLoading();
+
+      if (!shouldStartHidden) {
+        await applicationIsLoadingWindow.show();
+      }
+
+      await whenApplicationIsLoading();
```

Comment:
I found this name to be confusing since it was non-obvious that adding something to `applicationIsLoading` causes the application to take longer while loading.

I think we should change the nominclature a bit.

Reply:
Any suggestions?

Reply:
How about:

1. `beforeElectronIsReady` (no change)
2. `afterElectronIsReady`
4. `beforeMainWindowVisible`
5. `afterMainWindowVisibile`

Reply:
We changed name of `whenApplicationIsLoading` to `onLoadOfApplication` to better indicate that adding stuff here adds to stuff that happens on load of application.

Also, we renamed `applicationIsLoadingWindow` to `splashWindow` to better distinguish it from `onLoadOfApplication` which has different responsibilities.